tests/unit/nupic/algorithms/sdr_classifier_test.py

```diff
@@ -0,0 +1,445 @@
+    self.assertAlmostEqual(result[1][2], 0.034234, places=5)
+    self.assertAlmostEqual(result[1][3], 0.034234, places=5)
+    self.assertAlmostEqual(result[1][4], 0.093058, places=5)
+    self.assertAlmostEqual(result[1][5], 0.770004, places=5)
+
+
+  def testComputeWithMissingValue(self):
+    c = self._classifier([1], 0.1, 0.1, 0)
+    result = c.compute(
+        recordNum=0, patternNZ=[1, 5, 9],
+        classification={'bucketIdx': None, 'actValue': None},
+        learn=True, infer=True)
+
+    self.assertSetEqual(set(result.keys()), set(('actualValues', 1)))
+    self.assertEqual(len(result['actualValues']), 1)
+    self.assertEqual(result['actualValues'][0], None)
+
+
+  def testComputeCategory(self):
+    c = self._classifier([1], 0.1, 0.1, 0)
+    c.compute(recordNum=0, patternNZ=[1, 5, 9],
+              classification={'bucketIdx': 4, 'actValue': 'D'},
+              learn=True, infer=True)
+    result = c.compute(recordNum=1, patternNZ=[1, 5, 9],
+                       classification={'bucketIdx': 4, 'actValue': 'D'},
+                       learn=True, infer=True)
+    self.assertSetEqual(set(result.keys()), set(('actualValues', 1)))
+    self.assertEqual(result['actualValues'][4], 'D')
+
+    predictResult = c.compute(recordNum=2, patternNZ=[1, 5, 9],
+                              classification={'bucketIdx': 5,
+                                              'actValue': None},
+                              learn=True, infer=True)
+    for value in predictResult['actualValues']:
+      self.assertIsInstance(value, (types.NoneType, types.StringType))
+
+
+  def testComputeCategory2(self):
+    c = self._classifier([1], 0.1, 0.1, 0)
+    c.compute(recordNum=0, patternNZ=[1, 5, 9],
+              classification={'bucketIdx': 4, 'actValue': 'D'},
+              learn=True, infer=True)
+    result = c.compute(recordNum=1, patternNZ=[1, 5, 9],
+                       classification={'bucketIdx': 4, 'actValue': 'E'},
+                       learn=True, infer=True)
+    self.assertSetEqual(set(result.keys()), set(('actualValues', 1)))
+    self.assertEqual(result['actualValues'][4], 'D')
+
+
+  def testSerialization(self):
+    c = self._classifier([1], 1.0, 0.1, 0)
+    c.compute(recordNum=0,
+              patternNZ=[1, 5, 9],
+              classification={'bucketIdx': 4, 'actValue': 34.7},
+              learn=True, infer=True)
+    c.compute(recordNum=1,
+              patternNZ=[0, 6, 9, 11],
+              classification={'bucketIdx': 5, 'actValue': 41.7},
+              learn=True, infer=True)
+    c.compute(recordNum=2,
+              patternNZ=[6, 9],
+              classification={'bucketIdx': 5, 'actValue': 44.9},
+              learn=True, infer=True)
+    c.compute(recordNum=3,
+              patternNZ=[1, 5, 9],
+              classification={'bucketIdx': 4, 'actValue': 42.9},
+              learn=True, infer=True)
+    serialized = pickle.dumps(c)
+    c = pickle.loads(serialized)
+    result = c.compute(recordNum=4,
```

Comment:
We should also check that the member variables are set as expected after deserialization.  For example, alpha could be incorrect after unpickling, and this test would not catch that.